modules/nlu/src/backend/election/natural-election.ts

```diff
@@ -45,20 +48,47 @@ function electIntent(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstandin
   }
 }
 
-function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
-  if (!input.predictions) {
+export function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
+  if (!isNotNil(input.predictions)) {
     return input
   }
 
-  const elected = input.intent!
+  const elected = input.intent
+  if (!isNotNil(elected)) {
+    return input
+  }
 
-  if (!elected) {
+  const electedContext = input.predictions[elected.context]
+  if (!isContextPrediction(electedContext)) {
     return input
   }
 
-  const electedIntent = input.predictions[elected.context].intents.find(i => i.label === elected.name)
-  if (!electedIntent) {
+  const electedContextIntents = electedContext.intents
```

Comment:
This is the bug right ? If so, what was the `elected.context` value and why wasn't is in `input.predictions` ?

Reply:
Yes this is where the bug was. `input.predictions[elected.context]` could be undefined, throwing an error on property access `intents`.

The only **why** I can provide right now is the type-checking wasn't strong enough, which my PR now fixes. The **why** over that, as in why the input was like that, is outside the scope of this PR. Might not even be a problem since input can be returned as is.

Reply:
Still curious on what was the value of `elected.context` and why it was not in `input.predictions`

Reply:
With no intents registered for this bot:
![Screenshot from 2021-09-29 08-34-02](https://user-images.githubusercontent.com/30733005/135269238-40906622-101f-450e-ac8c-5764e2f5a5e4.png)
![Screenshot from 2021-09-29 08-33-31](https://user-images.githubusercontent.com/30733005/135269266-299a465e-f64a-4297-a523-47e750700621.png)